File: Programming-Basics-With-JavaScript/Homeworks-And-Labs/04.For-Loop/02-exercise/08.tennisRanklist.js
function tennisRanklist(input) {

    const pointsForW = 2000;
    const pointsForF = 1200;
    const pointsForSF = 720;

    let index = 0;
    let numberOfTournaments = Number(input[index]);
    index++;
    let startingPoints = Number(input[index]);
    index++;

    let tournamentPoints = 0;
    let wins = 0;

    for (let i = 1; i <= numberOfTournaments; i++) {
        let currentTournament = input[index];
        index++;

        switch (currentTournament) {
            case "W": tournamentPoints += pointsForW;
                wins++;
                break;
            case "F": tournamentPoints += pointsForF; break;
            case "SF": tournamentPoints += pointsForSF; break;
        }
    }

    let totalPoints = startingPoints + tournamentPoints;
    let averagePoints = tournamentPoints / numberOfTournaments;
    let winPercentage = wins / numberOfTournaments * 100;

    console.log(`Final points: ${totalPoints}`);
    console.log(`Average points: ${Math.floor(averagePoints)}`);
    console.log(`${winPercentage.toFixed(2)}%`);
}
tennisRanklist(["5", "1400", "F", "SF", "W", "W", "SF"]);

File: Python_Basics/Homeworks-And-Labs/04.for_loop/02.exercise/08.tennis_ranklist.py
from math import floor

points_for_w = 2000
points_for_f = 1200
points_for_sf = 720

number_of_tournaments = int(input())
starting_points = int(input())

tournament_points = 0
wins = 0

for i in range(number_of_tournaments):
    current_tournament = input()

    if current_tournament == 'W':
        tournament_points += points_for_w
        wins += 1
    elif current_tournament == 'F':
        tournament_points += points_for_f
    elif current_tournament == 'SF':
        tournament_points += points_for_sf

total_points = starting_points + tournament_points
average_points = tournament_points / number_of_tournaments
win_percentage = wins / number_of_tournaments * 100

print(f'Final points: {total_points}')
print(f'Average points: {floor(average_points)}')
print(f'{win_percentage:.2f}%')
